routers/web/repo/repo.go

```diff
@@ -597,3 +599,35 @@ func SearchRepo(ctx *context.Context) {
 		Data: results,
 	})
 }
+
+// getLatestCommitStatusState returns the commit status state returns the latest commit status state of the latest
+// commit on the default branch of the given repository
+func getLatestCommitStatusState(ctx *context.Context, repo *repo_model.Repository) api.CommitStatusState {
```

Comment:
Functionality-wise I'm all for this PR.
However, I can imagine some huge performance hits for larger repos/ instances…

Reply:
I didn't find a function that does this. In fact, I got lost reading so many functions from modules/models/services of git/repository/branch...

I'm doing weird stuff here like iterating over all branches when I only need the default branch. I hope someone with more knowledge in the codebase help make this better

Reply:
I'm no longer making `O(n)` database calls, only `O(1)`. Is that good enough? (caching could be a future improvement)

Reply:
O(1) is of course awesome 😆

Reply:
BTW as for caching, it should also be possible to build a mini-redis into SQL by abusing a table specifically for this purpose. So it may be possible to avoid a hard redis dependency.